docs/source/usage.rst

```diff
@@ -534,22 +534,33 @@ namespace, and to use interpolations in the name itself. The following example d
     4
 
 
-By default a custom resolver's output is cached, so that when it is called with the same
-inputs we always return the same value. This behavior may be disabled by setting ``use_cache=False``:
+By default a custom resolver is called on every access, but it is possible to cache its output
+by registering it with ``use_cache=True``.
+This may be useful either for performance reasons or to ensure the same value is always returned.
+Note that the cache is based on the string literals representing the resolver's inputs, and not
+the inputs themselves:
 
 .. doctest::
 
     >>> import random
     >>> random.seed(1234)
-    >>> OmegaConf.register_new_resolver("cached", random.randint)
     >>> OmegaConf.register_new_resolver(
-    ...              "uncached", random.randint, use_cache=False)
-    >>> c = OmegaConf.create({"cached": "${cached:0,10000}",
-    ...                       "uncached": "${uncached:0,10000}"})
+    ...    "cached", random.randint, use_cache=True
+    ... )
+    >>> OmegaConf.register_new_resolver("uncached", random.randint)
+    >>> c = OmegaConf.create(
+    ...     {
+    ...         "cached_1": "${cached:0,10000}",
+    ...         "uncached": "${uncached:0,10000}",
+    ...         "cached_2": "${cached:0,${uncached}}",
+    ...     }
+    ... )
     >>> # same value on repeated access thanks to the cache
-    >>> assert c.cached == c.cached == 7220
+    >>> assert c.cached_1 == c.cached_1 == 7220
     >>> # not the same since the cache is disabled
     >>> assert c.uncached != c.uncached
+    >>> # same string literal "${uncached}" => same value
+    >>> assert c.cached_2 == c.cached_2 == 1192
```

Comment:
I think we should also make the point that the caching is at the function call level and not at the node level:

```
"cached_1": "${cached:0,10000}",
"cached_3": "${cached:0,10000}",
```

cached_1 and cached_3 will always output the same value.

Reply:
Indeed, done in 71b5536 (I made it `cache_2` as I felt like it flowed better that way)

Reply:
> Indeed, done in [71b5536](https://github.com/omry/omegaconf/commit/71b553695cc4adeca48d3940817b4d835a3fc190) (I made it `cache_2` as I felt like it flowed better that way)

Yeah, I didn't really suggest that you use cached_3 for this one. cached_2 makes more sense :)